bin/exercise-gen.pl6

```diff
@@ -15,56 +13,62 @@ given $base-dir {
   }
 }
 
+my $yaml-filename := '.meta/exercise-data.yaml';
+
+my %*SUB-MAIN-OPTS = :named-anywhere;
+
 multi sub MAIN (Bool:D :$all where *.so) {
-  generate .basename for $base-dir.add('exercises').dir;
+  MAIN $base-dir.add('exercises').dir».basename;
 }
 
 multi sub MAIN (*@exercises) {
-  @exercises».&generate;
+  await @exercises.unique.map: {start .&generate};
 }
 
 multi sub MAIN {
-  say 'No args given; working in current directory.';
-  if '.meta/exercise-data.yaml'.IO ~~ :f {
-    generate $*CWD.IO.basename;
-  } else {
-    say 'exercise-data.yaml not found in .meta of current directory; exiting.';
-    exit;
+  given $*CWD {
+    say "No args given; looking for '$yaml-filename' in '$_'";
```

Comment:
Since this is a message to someone that is running, perhaps "No arguments given:" will be appropriate.